sycl/plugins/unified_runtime/ur/adapters/hip/context.cpp

```diff
@@ -106,7 +106,8 @@ urContextGetInfo(ur_context_handle_t hContext, ur_context_info_t propName,
 UR_APIEXPORT ur_result_t UR_APICALL
 urContextRelease(ur_context_handle_t hContext) {
   if (hContext->decrementReferenceCount() == 0) {
-    delete hContext;
+    hContext->invokeExtendedDeleters();
+    std::unique_ptr<ur_context_handle_t_> WillDeleteContext{hContext};
```

Comment:
Why not just `delete hContext` here, the unique pointer seems a little overkill

Reply:
The plugins all seem to use unique_ptrs to delete. I think it's good to have a consistent style but I think it's good to rename the `unique_ptr`s to `WillDeleteX` or whatever

Reply:
It's also safer presumably, in case the context wasn't allocated with `new` (maybe it was allocated with `make_unique` etc

Reply:
Yeah I didn't realize but apparently the spec says a call to `make_unique` needs to use `new` https://en.cppreference.com/w/cpp/memory/unique_ptr/make_unique This is not a hill I am willing to die on so I will remove this line